drivers/focuser/alluna_tcs2.cpp

```diff
@@ -844,33 +845,43 @@ bool AllunaTCS2::getDustCover()
     ParkCapSP[CAP_UNPARK].setState((value == 1) ? ISS_ON : ISS_OFF);
     ParkCapSP[CAP_PARK  ].setState((value != 1) ? ISS_ON : ISS_OFF);
     ParkCapSP.setState(IPS_OK);
+    ParkCapSP.apply();
 
     return true;
 }
 
 IPState AllunaTCS2::ParkCap()
 {
-    if (ParkCapSP[CAP_PARK].getState() == ISS_OFF)
+    if (ParkCapSP[CAP_PARK].getState() == ISS_ON)
     {
-        if (setDustCover()) // toggle state of dust cover
+        LOG_INFO("Toggle");
+        if (setDustCover()) { // toggle state of dust cover
+            isCoverMoving = true;
             return IPS_BUSY;
+        }
         else
             return IPS_ALERT;
     }
+    getDustCover();
 
     // Cover already parked, nothing to do
     return IPS_OK;
 }
 
 IPState AllunaTCS2::UnParkCap()
 {
-    if (ParkCapSP[CAP_UNPARK].getState() == ISS_OFF)
+    LOGF_INFO("UnParkCap called, state is %d %d", ParkCapSP[CAP_PARK].getState(), ParkCapSP[CAP_UNPARK].getState());
```

Comment:
Should be LOGF_DEBUG

Reply:
Yes, I forgot to remove some debugging code.

Reply:
Fixed